src/lib/HST/Application.hs

```diff
@@ -103,34 +111,72 @@ useAlgo ms = do
-collectDataDecl (S.DataDecl _ _ dataName conDecls) = do
-  let dataQName  = S.UnQual S.NoSrcSpan dataName
-      conEntries = map (makeConEntry dataQName) conDecls
-  modifyEnv
-    $ insertDataEntry DataEntry { dataEntryName = dataQName
-                                , dataEntryCons = map conEntryName conEntries
-                                }
-  mapM_ (modifyEnv . insertConEntry) conEntries
-collectDataDecl _ = return ()
+--   Returns @Nothing@ if the given declaration is not a data type declaration.
+createModuleInterfaceEntry :: S.Decl a -> Maybe (TypeName a, [ConEntry a])
+createModuleInterfaceEntry (S.DataDecl _ _ dataName conDecls)
+  = let dataQName  = S.UnQual S.NoSrcSpan dataName
+        conEntries = map (makeConEntry dataQName) conDecls
+    in Just (dataQName, conEntries)
+createModuleInterfaceEntry _ = Nothing
 
--- | Creates an environment entry for a constructor declaration.
+-- | Creates a module interface entry for a constructor declaration.
 makeConEntry :: S.QName a -> S.ConDecl a -> ConEntry a
 makeConEntry dataQName conDecl = ConEntry
   { conEntryName    = S.UnQual S.NoSrcSpan (S.conDeclName conDecl)
   , conEntryArity   = S.conDeclArity conDecl
   , conEntryIsInfix = S.conDeclIsInfix conDecl
   , conEntryType    = dataQName
   }
+
+-------------------------------------------------------------------------------
+-- Environment Initialization                                                --
+-------------------------------------------------------------------------------
+-- | Initializes the environment for the module stored at the given file path.
+--
+--   The module interfaces of imported modules are added to the environment.
+--   If such a module is unavailable, the import is skipped and the user is
+--   informed about the skip.
+initializeEnvironment
+  :: Members '[InputModule a, Report] r => FilePath -> Sem r (Environment a)
+initializeEnvironment filePath = do
+  S.Module _ _ _ imports _ <- getInputModule filePath
+  currentModule <- getInputModuleInterface filePath
+  mInterfaces <- mapM (getInputModuleInterfaceByName . S.importModule) imports
+  mImportedModules <- zipWithM reportMissingModule imports mInterfaces
+  return Environment { envCurrentModule   = currentModule
+                     , envImportedModules = catMaybes mImportedModules
+                     , envOtherEntries    = preludeModuleInterface
+                     }
+ where
+  -- | Returns the given import declaration and module interface, if the module
+  --   could be successfully be imported, or reports the skip of the missing
+  --   module and returns @Nothing@.
+  reportMissingModule :: Member Report r
+                      => S.ImportDecl a
+                      -> Maybe (ModuleInterface a)
+                      -> Sem r (Maybe (S.ImportDecl a, ModuleInterface a))
+  reportMissingModule importDecl Nothing                = do
+    report
+      $ message Info (S.importSrcSpan importDecl)
```

Comment:
If a module could not be found, that's actually quite a severe error. I think that it should be at least a warning.

```suggestion
      $ message Warning (S.importSrcSpan importDecl)
```

Reply:
The user could have imported a module that hasn't been given as an input file to the pattern matching compiler. If only functions from such a module are used, this isn't even an error at all, at least in my opinion.
I can of course still change the severity to a `Warning`.